packages/translate/lib/testLineResult.tsx

```diff
@@ -362,106 +365,125 @@ const translateInvalidRepositoryReference = (result: InvalidRepositoryReferenceE
 	? baseScreenshotPath + screenshotPath
 	: undefined;
 
+const getInvertedResultMessage = (
+	testLine: TestLine,
+	lineResult?: TestLineResult
+): Node | undefined => {
+	if (
+		!lineResult // Does not make sense to translate inverted result if there is no result to compare to
+		|| (testLine.type !== 'assert' && testLine.type !== 'wait') // Only assert lines can be inverted by definition
+		|| !('then' in testLine) // Make the code future-proof, we might drop inverse conditions
+		|| testLine.then === 'success' // Not an inverse result
+		|| (lineResult.errorType && lineResult.errorType !== 'queryFailed' && lineResult.errorType !== 'appRunning') // failure not related to condition, thus can't be inverted
+	) {
+		return undefined;
+	}
+
+	return lineResult.result === testLine.then ? conditionWasMetMessage : conditionWasNotMetMessage;
+};
+
+const getLineResultMessage = (testLine: TestLine, lineResult?: TestLineResult): Node | undefined => {
+	const invertedResult = getInvertedResultMessage(testLine, lineResult);
+	if (invertedResult) {
+		return invertedResult;
+	}
+
+	// Handle case when result is success
+	if (!lineResult || lineResult.result === 'success') {
+		return undefined;
+	}
+
+	if (lineResult.result === 'excluded') {
+		return <text>Line was not executed</text>;
+	}
+
+	return translateResultErrorMessage(lineResult as TestLineErrorResult);
+};
+
 export const translateTestLineResult = (options: {
 	testLine: TestLine,
 	appConfig: AppConfiguration,
 	lineResult?: TestLineResult,
 	elements?: Elements,
 	snippets?: Snippets,
 }): TestLineResultNode => {
-	const testLineTranslation = translateTestLine(options);
 	const {lineResult} = options;
-	const {then} = options.testLine as AssertTestLine;
-
-	if (lineResult && then !== 'success') {
-		const messege = lineResult?.result === 'success'
-			? <text>Condition was not met</text>
-			: <text>Condition was met</text>;
-		const unexpectedResult = lineResult.errorType && !['appRunning', 'queryFailed'].includes(lineResult.errorType);
-
-		return <test-line-result
-			status={lineResult?.result}
-			message={unexpectedResult ? translateResultMessage(lineResult as TestLineErrorResult) : messege}
-		>{testLineTranslation}</test-line-result> as TestLineResultNode;
-	}
-
-	if (!lineResult || lineResult.result === 'success' || lineResult.result === 'excluded' || !('errorType' in lineResult)) {
-		// TODO: not pass "success" to status if lineResult is undefined
-		return <test-line-result
-			status={lineResult?.result ?? 'success'}
-			screenshot={getScreenshotUrl(lineResult?.screenshot)}
-		>{testLineTranslation}</test-line-result> as TestLineResultNode;
-	}
 
 	return <test-line-result
-		status={lineResult.result}
-		message={translateResultMessage(lineResult)}
-		screenshot={getScreenshotUrl(lineResult.screenshot)}
-	>{testLineTranslation}</test-line-result> as TestLineResultNode;
+		status={lineResult?.result ?? 'success'}
```

Comment:
Line result without result will be translated as success?

Reply:
Yes. If you don't want result - use `testLine`, not `testLineResult`.